File: modules/examples/src/web_workers/word_tree/tagger.dart
library examples.src.web_workers.word_tree.tagger;

import 'dart:js';

List<dynamic> tagWords(String text) {
  var lexer = new JsObject(context['Lexer']);
  var words = lexer.callMethod('lex', [text]);

  var tagger = new JsObject(context['POSTagger']);
  var taggedWords = tagger.callMethod('tag', [words]);

  return taggedWords;
}

File: modules/examples/src/web_workers/word_tree/tagger.ts
import {List} from 'angular2/src/facade/collection';

// TODO: Make a typings file for these and import properly
declare var POSTagger;
declare var Lexer;

export function tagWords(text: string): List<any> {
  var words = new Lexer().lex(text);
  var taggedWords = new POSTagger().tag(words);
  return taggedWords;
}
